article/serializers.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import serializers
+from .models import (
+    Article,
+    ArticleComment,
+    ArticleVote
+)
+from users.serializers import UserProfileSerializer , UserSerializer
+
+
+
+class ArticleSerializer(serializers.ModelSerializer):
+    user = serializers.SerializerMethodField(read_only=True)
+    
+    class Meta:
+        model = Article
+        fields = '__all__'
+
+    def get_user(self, obj):
+        user = obj.user.userprofile
+        serializer = UserProfileSerializer(user, many=False)
+        return serializer.data
+
+class ArticleCommentSerializer(serializers.ModelSerializer):
```

Comment:
class ArticleCommentSerializer(serializers.ModelSerializer):
    class Meta:
        model = ArticleComment
        fields = '__all__'
        read_only_fields= ("user",)

Reply:
read_only=True did the same